app/uk/gov/hmrc/cataloguefrontend/connector/BuildDeployApiConnector.scala

```diff
@@ -176,7 +176,30 @@ class BuildDeployApiConnector @Inject() (
 
     signAndExecuteRequest(
       endpoint = "CreateRepository",
-      body     = body
+      body = body
+    ).map(_.map(resp => resp.details.as[AsyncRequestId]))
+  }
+
+  def createAPrototypeRepository(payload: CreatePrototypeRepoForm): Future[Either[String, AsyncRequestId]] = {
+    val finalPayload =
+      s"""
+         |{
+         |   "push_template_repo": "true",
+         |   "repository_name": "${payload.repositoryName}",
+         |   "password": ${payload.password},
+         |   "team_name": "${payload.teamName}",
+         |   "init_webhook_version": "2.2.0",
+         |   "default_branch_name": "main",
+         |   "slack_notification_channels": "${payload.slackChannels}"
+         |}""".stripMargin
+
+    val body = Json.parse(finalPayload)
+
+    logger.info(s"Calling the B&D Create Prototype Repository API with the following payload: ${body}")
```

Comment:
This would leak the prototype password in logs I think

Reply:
Yes if info gets switched on in prod so have now obfuscated it.